### PR TITLE
wd/sec: add AEAD algorithm driver for uadk

### PR DESCRIPTION
**Pull request by Liulongfang**
1.Added AEAD APIs for uadk
2.Added AEAD user mode driver for uadk
3.Fixed some coding styles

Signed-off-by: Liulongfang <836713992@qq.com>

### CODE REVIEW

File: drv/hisi_sec.c [hunk 1]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
 };
 
 WD_DIGEST_SET_DRIVER(hisi_digest_driver);
+
+static int fill_aead_bd2_alg(struct wd_aead_msg *msg,
+		struct hisi_sec_sqe *sqe)
```
Comment (wangzhou) [inline drv/hisi_sec.c:764]: coding style

File: drv/hisi_sec.c [hunk 2]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+{
+#define IV_LAST_BYTE1		1
+#define IV_LAST_BYTE2		2
+#define IV_CTR_INIT		1
+#define IV_CM_CAL_NUM		2
+#define IV_CL_MASK		0x7
+#define IV_FLAGS_OFFSET	0x6
+#define IV_CM_OFFSET		0x3
+#define IV_LAST_BYTE2_MASK	0xFF00
+#define IV_LAST_BYTE1_MASK	0xFF
```
Comment (wangzhou) [inline drv/hisi_sec.c:877]: please move these at the head of file

File: drv/hisi_sec.c [hunk 3]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+		flags |= cl;
+
+		/* the M' is bit3~bit5, the Flags is bit6 */
+		cm = (msg->auth_bytes - IV_CM_CAL_NUM) / IV_CM_CAL_NUM;
+		flags |= cm << IV_CM_OFFSET;
+		if (msg->assoc_bytes > 0)
+			flags |= 0x01 << IV_FLAGS_OFFSET;
+
+		msg->aiv[0] = flags;
+		/*
```
Comment (wangzhou) [inline drv/hisi_sec.c:899]: coding style

File: drv/hisi_sec.c [hunk 4]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+}
+
+int hisi_sec_aead_send(handle_t ctx, struct wd_aead_msg *msg)
+{
+	handle_t h_qp = (handle_t)wd_ctx_get_priv(ctx);
+	struct hisi_sec_sqe sqe;
+	__u8 scene, cipher, de, auth;
+	__u16 count = 0;
+	int ret;
+
```
Comment (wangzhou) [inline drv/hisi_sec.c:948]: do we need to check !ctx ?

File: drv/hisi_sec.c [hunk 5]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+		sqe.sds_sa_type = WD_DIGEST_THEN_CIPHER;
+	} else {
+		WD_ERR("failed to check aead op type!\n");
+		return -EINVAL;
+	}
+	sqe.sds_sa_type |= (__u8)(de | scene);
+	sqe.type_auth_cipher |= cipher;
+
+	if (msg->in_bytes == 0 ||
+		msg->in_bytes > MAX_INPUT_DATA_LEN) {
```
Comment (wangzhou) [inline drv/hisi_sec.c:977]: coding style

File: drv/hisi_sec.c [hunk 6]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+		WD_ERR("failed to fill aead bd mode!\n");
+		return ret;
+	}
+
+	fill_aead_bd2_addr(msg, &sqe);
+
+#ifdef DEBUG
+	WD_ERR("Dump aead send sqe-->!\n");
+	sec_dump_bd((unsigned char *)&sqe, SQE_BYTES_NUMS);
+#endif
```
Comment (wangzhou) [inline drv/hisi_sec.c:1002]: please remove these debug code

File: drv/hisi_sec.c [hunk 7]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+	if (ret < 0) {
+		WD_ERR("hisi qm send is err(%d)!\n", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static void parse_aead_bd2(struct hisi_sec_sqe *sqe, struct wd_aead_msg *
+recv_msg)
```
Comment (wangzhou) [inline drv/hisi_sec.c:1015]: coding style

File: drv/hisi_sec.c [hunk 8]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+	} else {
+		recv_msg->result = WD_SUCCESS;
+	}
+
+	recv_msg->tag = sqe->type2.tag;
+
+#ifdef DEBUG
+	WD_ERR("Dump aead recv sqe-->!\n");
+	sec_dump_bd((unsigned char *)sqe, SQE_BYTES_NUMS);
+#endif
```
Comment (wangzhou) [inline drv/hisi_sec.c:1033]: please remove this and any if you have

File: drv/hisi_sec.c [hunk 9]
```diff
@@ -756,3 +759,304 @@ static struct wd_digest_driver hisi_digest_driver = {
...
+		.drv_name	= "hisi_sec2",
+		.alg_name	= "aead",
+		.init		= hisi_sec_init,
+		.exit		= hisi_sec_exit,
+		.aead_send	= hisi_sec_aead_send,
+		.aead_recv	= hisi_sec_aead_recv,
+};
+
+WD_AEAD_SET_DRIVER(hisi_aead_driver);
+
```
Comment (wangzhou) [inline drv/hisi_sec.c:1062]: please remove this blank line

File: include/drv/wd_aead_drv.h [hunk 10]
```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+#ifndef __WD_AEAD_DRV_H
+#define __WD_AEAD_DRV_H
+
+#include "include/wd_alg_common.h"
+#include "include/wd_aead.h"
```
Comment (wangzhou) [inline include/drv/wd_aead_drv.h:6]: please remove "include"

File: include/drv/wd_aead_drv.h [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+/* fixme wd_aead_msg */
+struct wd_aead_msg {
```
Comment (wangzhou) [inline include/drv/wd_aead_drv.h:9]: why we need this fix me?

File: include/wd_aead.h [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+
+/**
+ * wd_cipher_type - Algorithm type of cipher
+ * statement in wd_cipher.h
+ *
+ * wd_cipher_mode - Algorithm mode of cipher
+ * statement in wd_cipher.h
+ */
+
+struct wd_aead_sess_setup {
```
Comment (wangzhou) [inline include/wd_aead.h:30]: add descriptions

File: include/wd_aead.h [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+	void			*iv;
+
+	__u32			in_bytes;
+	__u32			out_bytes;
+	__u32			out_buf_bytes;
+	__u16			iv_bytes;
+	__u16			assoc_bytes;
+	__u16			state;
+	wd_alg_aead_cb_t	*cb;
+	void			*cb_param;
```
Comment (wangzhou) [inline include/wd_aead.h:67]: add descriptions for each fields

File: include/wd_aead.h [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+};
+
+/**
+ * wd_aead_init() Initialise ctx configuration and schedule.
+ * @ config	    User defined ctx configuration.
+ * @ sched	    User defined schedule.
+ */
+int wd_aead_init(struct wd_ctx_config *config, struct wd_sched *sched);
+void wd_aead_uninit(void);
```
Comment (wangzhou) [inline include/wd_aead.h:76]: add commens

File: include/wd_aead.h [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+ * wd_aead_free_sess()
+ * @ sess, need to be freed sess
+ */
+void wd_aead_free_sess(handle_t h_sess);
+
+/**
+ * wd_aead_set_ckey() Set cipher key to aead session.
+ * @h_sess: wd aead session.
+ * @key: cipher key addr.
+ * @key_len: cipher key length.
```
Comment (wangzhou) [inline include/wd_aead.h:94]: add return value decription, and also for other return value

File: include/wd_aead.h [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+ */
+int wd_aead_set_akey(handle_t h_sess, const __u8 *key, __u16 key_len);
+
+/**
+ * wd_do_aead_sync()/ async() Syn/asynchronous aead operation
+ * @sess: wd aead session
+ * @req: operational data.
+ */
+int wd_do_aead_sync(handle_t h_sess, struct wd_aead_req *req);
+int wd_do_aead_async(handle_t h_sess, struct wd_aead_req *req);
```
Comment (wangzhou) [inline include/wd_aead.h:112]: add description

File: include/wd_aead.h [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+ * wd_aead_get_authsize() Get authenticate data length from aead session.
+ * @h_sess: wd aead session.
+ */
+int wd_aead_get_authsize(handle_t h_sess);
+
+/**
+ * wd_aead_get_maxauthsize() Get max authenticate data length from aead API.
+ * @h_sess: wd aead session.
+ */
+int wd_aead_get_maxauthsize(handle_t h_sess);
```
Comment (wangzhou) [inline include/wd_aead.h:131]: same here

File: include/wd_aead.h [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+ * @h_sess: wd aead session.
+ * @authsize: authenticate data length.
+ */
+int wd_aead_set_authsize(handle_t h_sess, __u16 authsize);
+
+/**
+ * wd_aead_get_authsize() Get authenticate data length from aead session.
+ * @h_sess: wd aead session.
+ */
+int wd_aead_get_authsize(handle_t h_sess);
```
Comment (wangzhou) [inline include/wd_aead.h:125]: can  here be __u32?

File: include/wd_aead.h [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+ * wd_aead_poll_ctx() poll operation for asynchronous operation
+ * @index: index of ctx which will be polled.
+ * @expt: user expected num respondings
+ * @count: how many respondings this poll has to get.
+ */
+int wd_aead_poll_ctx(__u32 index, __u32 expt, __u32* count);
+
+/**
+ * wd_aead_poll() Poll finished request.
+ * this function will call poll_policy function which is registered to wd aead
```
Comment (wangzhou) [inline include/wd_aead.h:143]: add description for expt and count

File: drv/hisi_sec.c (hunk 4 above)
Comment (Liulongfang) [reply drv/hisi_sec.c:948]: ctx don't need check

File: drv/hisi_sec.c (hunk 6 above)
Comment (Liulongfang) [reply drv/hisi_sec.c:1002]: this debug code is necessary for cipher, digest and aead

File: drv/hisi_sec.c (hunk 3 above)
Comment (Liulongfang) [reply drv/hisi_sec.c:899]: there is no coding style

File: include/drv/wd_aead_drv.h (hunk 10 above)
Comment (Liulongfang) [reply include/drv/wd_aead_drv.h:6]: I have kept the style consistent, no need to modify here

File: include/wd_aead.h (hunk 12 above)
Comment (Liulongfang) [reply include/wd_aead.h:30]: You only need to add a comment when the meaning is not clear. 
The meaning here is very straightforward and no comment is required

File: include/wd_aead.h (hunk 15 above)
Comment (Liulongfang) [reply include/wd_aead.h:94]: The return value description is in the interface document, no need to repeat the description here

File: include/wd_aead.h (hunk 18 above)
Comment (Liulongfang) [reply include/wd_aead.h:125]: can't

File: drv/hisi_sec.c (hunk 2 above)
Comment (Liulongfang) [reply drv/hisi_sec.c:877]: There is no need to move to the file header，them just used in this function

File: include/wd_aead.h (hunk 16 above)
Comment (Liulongfang) [reply include/wd_aead.h:127]: Two functions have been commented together

**Comment by Liulongfang** [comment]
modify it

**Pull request closed without merging**